rbac/management/role/relation_api_dual_write_handler.py

```diff
@@ -120,36 +125,76 @@ def replicate_deleted_system_role(self):
         self._replicate(
             ReplicationEventType.DELETE_SYSTEM_ROLE,
             self._create_metadata_from_role(),
-            self._generate_relations_for_role(),
+            self._generate_relations_for_role(list_all_possible_scopes=True),
             [],
         )
 
-    def _generate_relations_for_role(self) -> list[common_pb2.Relationship]:
+    def check_create_admin__platform_relation(self, role, role_scope, list_relations, permissions):
```

Comment:
Is there a reason for the double underscore here? Also, this should probably start with an underscore to signify that it's internal.